### DISCUSSION TITLE
How eject theme

### DISCUSSION THREAD

**Discussion by JulioJu** [vuepress/core#56] [Q&A] [answered]
Hi,

I would like to create my own custom theme, but when I copy vuepress-next/packages/@vuepress/theme-default/src/ into my project I have the error shown below.

My project is at https://github.com/JulioJu/blog . It is very little. 

I've seen that the `vuepress eject` command was removed. I don't find any solution. I've take a long time to try different solutions without success.

I guess it's not possible currently. 

```
Uncaught (in promise) Error: useSidebarItems() is called without provider.
    useSidebarItems useSidebarItems.ts:37
    setup Layout.vue:41
    callWithErrorHandling runtime-core.esm-bundler.js:154
    setupStatefulComponent runtime-core.esm-bundler.js:6419
    setupComponent runtime-core.esm-bundler.js:6380
    mountComponent runtime-core.esm-bundler.js:4118
    processComponent runtime-core.esm-bundler.js:4094
    patch runtime-core.esm-bundler.js:3712
    componentEffect runtime-core.esm-bundler.js:4278
    reactiveEffect reactivity.esm-bundler.js:42
    callWithErrorHandling runtime-core.esm-bundler.js:154
    flushJobs runtime-core.esm-bundler.js:362
```

Thanks in advance :smile:

**Comment by yzx9** [comment, marked as answer]
The command `vuepress eject` was [removed](https://github.com/vuepress/vuepress-next#cli), you can try to use `extends`.

`useSidebarItems` required provider injected from `.clientAppSetup.ts`, however, it won't work in your case because the default theme will released after compile to javascript but you are using it directly. So you can update your theme config to fix this error like below:

Now: 
```typescript
// docs/theme/config.ts
clientAppEnhanceFiles: path.resolve(__dirname, './clientAppEnhance.js'),
clientAppSetupFiles: path.resolve(__dirname, './clientAppSetup.js'),
```

Change the file suffix `.js` to `.ts`:
```typescript
// docs/theme/config.ts
clientAppEnhanceFiles: path.resolve(__dirname, './clientAppEnhance.ts'),
clientAppSetupFiles: path.resolve(__dirname, './clientAppSetup.ts'),
```

**Reply by JulioJu** [reply]
Thanks a lot :-) .

Sorry for the delay to say you thank you, I thought I've already said thanks for your detailed answer .
